reinforcement_learning/common/sagemaker_rl/ray_launcher.py

```diff
@@ -231,6 +240,47 @@ def save_checkpoint_and_serving_model(self, algorithm=None, env_string=None, con
         # To ensure SageMaker local mode works fine
         change_permissions_recursive(INTERMEDIATE_DIR, 0o777)
         change_permissions_recursive(MODEL_OUTPUT_DIR, 0o777)
+        
+    def set_up_checkpoint(self, config=None):
+        try:
+            checkpoint_dir = config['training']['restore']
+            print("Found checkpoint dir in user config.")
+        except KeyError:
+            pass
+
+        try:
+            # validate files in checkpoint channel and set restore dir in ray config
+            validation = 0
+            if len(os.listdir(CHECKPOINT_DIR)) is not (2 or 3):
+                raise RuntimeError("Unexpected files in checkpoint dir.",
+                                   "Please check ray documents for the correct checkpoint format.")
+            checkpoint_dir = ""
+            for filename in os.listdir(CHECKPOINT_DIR):
+                is_tune_metadata= filename.endswith("tune_metadata")
+                is_extra_data = filename.endswith("extra_data")
+                is_checkpoint_meta = is_tune_metadata + is_extra_data
+                validation += is_checkpoint_meta
+                if not is_checkpoint_meta:
```

Comment:
Once come into a valid checkpoint file, this condition will never fail after that